octavia_f5/controller/worker/tasks/f5_tasks.py

```diff
@@ -272,10 +278,75 @@ def execute(self, bigip: bigip_restclient.BigIPRestClient,
         return device_route
 
 
+class SyncSubnetRoutes(task.Task):
+    """ Task to create static subnet routes """
+
+    @decorators.RaisesIControlRestError()
+    def execute(self, bigip: bigip_restclient.BigIPRestClient,
+                selfips: [network_models.Port],
+                network: f5_network_models.Network):
+
+        # Skip passive device if route_on_active is enabled
+        if CONF.networking.route_on_active and not bigip.is_active:
+            return None
+
+        def subnet_in_selfips(subnet, selfips):
+            for selfip in selfips:
+                for fixed_ip in selfip.fixed_ips:
+                    if fixed_ip.subnet_id == subnet:
+                        return True
+            return False
+
+        # Fetch existing routes
+        response = bigip.get(path=f"/mgmt/tm/net/route?$filter=partition+eq+Common").json()
+        existing_routes = response.get('items', [])
+
+        # subnet routes that must exist (routes already exist for SelfIPs)
+        subnets_that_need_routes = [subnet for subnet in network.subnets if not subnet_in_selfips(subnet, selfips)]
+
+        # delete existing subnet routes that aren't needed anymore - we'll only provision the missing ones
+        for existing_route in existing_routes:
+            existing_route_name = existing_route['name']
+
+            # ignore routes that are not subnet routes of this network
+            subnet_route_network_part = get_subnet_route_name(network.id, '')
+            if not existing_route_name.startswith(subnet_route_network_part):
+                continue
+
+            # delete the route or ignore it if it's still needed
+            delete_device_route = True
+            for subnet_that_needs_route in subnets_that_need_routes:
+
+                # if the route is a needed subnet route, don't touch it
+                if existing_route_name == get_subnet_route_name(network.id, subnet_that_needs_route):
+                    subnets_that_need_routes.remove(subnet_that_needs_route)
```

Comment:
This is a very bad style. You don't want to modify the array you're currently iterating.
You can easily rewrite it by building a list of subnet-route-names and iterate through the disjoint elements.

Reply:
Good find.
Modifying the array we're currently iterating over is only risky as soon as we enter the next iteration. Right now I'm gravitating to simply putting a `break` statement after line 323 to make sure this is the last iteration, since at this point we're already sure we can ignore the subnet.
Creating a whole new array just for iteration is a bit too heavy in this specific instance, as compared to a `break`.